arrow/arrow.mojo

```diff
@@ -44,13 +44,16 @@ struct Bitmap:
     fn __len__(self) -> Int:
         return self.length
 
+    fn __del__(owned self):
+        self.data.free()
+
 
 struct ArrowBoolArray:
     var validity: Bitmap
     var buffer: Bitmap
 
 
-struct ArrowFixedWidthBuffer[T: AnyRegType]:
+struct ArrowFixedWidthBuffer[T: AnyTrivialRegType](AnyType):
```

Comment:
@sa- AnyRegType doesn't exist in 24.4. but when i try compile this I get.

![image](https://github.com/mojo-data/arrow.mojo/assets/4902386/e521e4a5-84c9-461b-af6d-b5691000671a)

Reply:
Since the C Data Interface is not being used by anything else you can comment out the release callback for now, and we can uncomment it when we would like to get the C Data Interface working